### PR TITLE
Add Carp

### PR DESCRIPTION
**Pull request by wltsmrz**
Add Carp language

## Checklist:

- [x] **I am adding a new language.**
  - [x?] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -6648,3 +6648,13 @@ xBase:
   tm_scope: source.harbour
   ace_mode: text
   language_id: 421
+Carp:
+  type: programming
+  ace_mode: clojure
+  codemirror_mode: clojure
+  codemirror_mime_type: text/x-clojure
+  color: "#c63c24"
```
Comment (bjorn3) [inline lib/linguist/languages.yml:6656]: For convenience: The color is `#c63c24`

Comment (wltsmrz) [reply lib/linguist/languages.yml:6656]: What's the markdown for embedding color?

**Comment by Alhadis** [comment]
@wltsmrz Please fill out the pull-request template, and be sure to have familiarised yourself with [the criteria for adding a new language](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language). Note that if the language lacks sufficient real-world usage, it's  ineligible for addition until it does. Judging from the [project's readme](https://github.com/carp-lang/Carp/blob/0faf8641e67040443c61f211579a64a784d99e81/README.md),

> _WARNING! This is a research project and a lot of information here might become outdated and misleading without any explanation. Don't use it for anything important just yet!_

I'd say Carp has yet to achieve the minimum required distribution of hundreds of users.

Please consult [`CONTRIBUTING.md`](https://github.com/github/linguist/blob/master/CONTRIBUTING.md) to learn more.

**Comment by wltsmrz** [comment]
> I'd say Carp has yet to achieve the minimum required distribution of hundreds of users.

I'll see if I can find out how many unique Carp repos there are on github, I'm curious myself.

**Comment by dx67hrod0lv** [comment]
My search topped out at 114 unique repositories.

As a reference, I compared the numbers I was getting for Carp to MoonScript, which tops out at 213 despite being added to linguist Jan 2013 and likely benefiting in terms of popularity from the free marketing of github tag. Although it may be a close call IMO, perhaps it doesn't meet today's popularity requirements.

See you down the road then?!

**Comment by wltsmrz** [comment]
It was brought to my attention that there was already a PR to add Carp https://github.com/carp-lang/Carp/issues/972#issuecomment-728919176

Although there are currently 114 repos I'll close this in favor of bumping the popularity tracking issue (https://github.com/github/linguist/issues/4219)

**Pull request closed without merging**